### DISCUSSION TITLE
dynamic speed

### DISCUSSION THREAD

**Discussion by maoxing** [curl/curl#13062] [Q&A] [answered]
i have multiple resources to download and use multi mode to create multiple handles to download. 

i need to control the download speed and adjust the speed at any time even when the download starts. 

What should I do in this case?

**Comment by dfandrich** [comment, marked as answer]
You can install progress callbacks and use whatever speed algorithm you want in there. Use `curl_easy_pause` to stop/stop transfers on each handle as your algorithm sees fit.

**Reply by maoxing** [reply]
I will create many handles, and these handles will be added through curl_multi_add_handle and set CURLMOPT_MAX_TOTAL_CONNECTIONS to 15. 

what I want to do is to be able to pause and resume the download of all handles at any time and adjust the download speed of the currently active handle at any time.

can you give some specific samples or experiences?

thanks

**Reply by bagder** [reply]
It sounds like you need `curl_easy_pause` and `CURLOPT_MAX_RECV_SPEED_LARGE`.

**Reply by maoxing** [reply]
can i control pause and resume via progress callback? 

If pause a handle, its progress callback continue to be called?

**Comment by bagder** [comment]
You can also update `CURLOPT_MAX_RECV_SPEED_LARGE` during transfer, and libcurl will adapt to that.
